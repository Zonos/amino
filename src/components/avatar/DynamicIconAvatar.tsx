import { DynamicIcon, DynamicIconType } from 'src/icons/icon-base/DynamicIcon';

import { AvatarBase, AvatarProps, iconSizes } from './AvatarBase';

export type DynamicIconAvatarProps = {
  icon: DynamicIconType;
} & AvatarProps;

export const DynamicIconAvatar = ({
  icon,
  shape,
  size,
}: DynamicIconAvatarProps) => (
  <AvatarBase shape={shape} size={size}>
    <DynamicIcon type={icon} size={iconSizes[size]} />
  </AvatarBase>
);
