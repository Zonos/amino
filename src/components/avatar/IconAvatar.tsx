import { ReactNode } from 'react';

import styled from 'styled-components';

import { AvatarBase, AvatarProps } from './AvatarBase';

export type IconAvatarProps = {
  icon: ReactNode;
} & AvatarProps;

const StyledAvatarBase = styled(AvatarBase)<Omit<IconAvatarProps, 'icon'>>`
  svg {
    width: 50%;
    height: 50%;
  }
`;

export const IconAvatar = ({
  icon,
  shape = 'round',
  size = 32,
  bordered = false,
}: IconAvatarProps) => (
  <StyledAvatarBase shape={shape} size={size} bordered={bordered}>
    {icon}
  </StyledAvatarBase>
);
