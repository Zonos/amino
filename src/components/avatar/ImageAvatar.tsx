import {
  AvatarBase,
  type FullAvatarProps,
} from 'src/components/avatar/AvatarBase';

export type ImageAvatarProps = {
  imageUrl: string;
} & FullAvatarProps;

/**
 * ImageAvatar displays an image in a standardized avatar format with consistent sizing and styling.
 * Use this component when you need to show profile pictures, product images, or any custom image
 * in an avatar format.
 *
 * @component Avatar
 *
 * @example Basic usage
 * ```tsx
 * <ImageAvatar
 *   imageUrl="https://example.com/profile.jpg"
 * />
 * ```
 *
 * @example With custom size and shape
 * ```tsx
 * <ImageAvatar
 *   imageUrl="https://example.com/profile.jpg"
 *   size={48}
 *   shape="square"
 *   bordered
 * />
 * ```
 *
 * @example Multiple sizes in a user profile
 * ```tsx
 * <VStack gap={16} alignItems="center">
 *   <ImageAvatar
 *     imageUrl="https://example.com/profile.jpg"
 *     size={64}
 *     bordered
 *   />
 *   <Text type="heading">Jane Smith</Text>
 *   <Text type="subheader" color={theme.gray600}>Product Manager</Text>
 * </VStack>
 * ```
 *
 * @example Image avatars in a horizontal list
 * ```tsx
 * <HStack gap={8}>
 *   {users.map(user => (
 *     <ImageAvatar
 *       key={user.id}
 *       imageUrl={user.profileImage}
 *       size={32}
 *       shape="rounded"
 *     />
 *   ))}
 * </HStack>
 * ```
 */
export const ImageAvatar = ({
  bordered = false,
  className,
  imageUrl,
  shape = 'round',
  size = 32,
  style,
}: ImageAvatarProps) => (
  <AvatarBase
    backgroundColor="transparent"
    backgroundSize="contain"
    backgroundUrl={imageUrl}
    bordered={bordered}
    className={className}
    shape={shape}
    size={size}
    style={style}
  />
);
