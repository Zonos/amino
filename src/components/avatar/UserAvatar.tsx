import {
  AvatarBase,
  type FullAvatarProps,
} from 'src/components/avatar/AvatarBase';

// eslint-disable-next-line no-relative-import-paths/no-relative-import-paths
import avatarImg from '../../../svgReact/icons/custom-svgs/Avatar.svg';

export type UserAvatarProps = FullAvatarProps;

/**
 * UserAvatar displays a default user placeholder avatar.
 * Use this component when you need to show a user avatar but don't have a specific image,
 * such as for anonymous users or as a fallback when user images are unavailable.
 *
 * @component Avatar
 *
 * @example Basic usage
 * <UserAvatar />
 *
 * @example Custom size with border
 * <UserAvatar
 *   size={48}
 *   bordered
 * />
 *
 * @example Different shapes
 * <HStack gap={16}>
 *   <UserAvatar shape="round" />
 *   <UserAvatar shape="rounded" />
 *   <UserAvatar shape="square" />
 * </HStack>
 *
 * @example In a user card
 * <Card>
 *   <VStack gap={8} alignItems="center" padding={16}>
 *     <UserAvatar size={64} bordered />
 *     <Text type="heading">Guest User</Text>
 *     <Text type="subheader" color={theme.gray600}>Unregistered</Text>
 *   </VStack>
 * </Card>
 */
export const UserAvatar = ({
  bordered = false,
  className,
  shape = 'round',
  size = 32,
  style,
}: UserAvatarProps) => (
  <AvatarBase
    backgroundPosition="bottom"
    backgroundSize="80%"
    backgroundUrl={`${avatarImg}`}
    bordered={bordered}
    className={className}
    shape={shape}
    size={size}
    style={style}
  />
);
