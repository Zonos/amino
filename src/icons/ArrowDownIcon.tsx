import React from 'react';

import { IconProps } from 'types';
import { IconBase } from './IconBase';

export const ArrowDownIcon = ({ size, color }: IconProps) => (
  <IconBase size={size} color={color}>
    <path
      d="M8 13L4 9m4 4l4-4m-4 4V3"
      stroke="currentColor"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </IconBase>
);
