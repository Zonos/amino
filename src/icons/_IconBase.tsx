import React, { forwardRef, ReactNode } from 'react';

import { type IconProps } from 'src/types/IconProps';
import styled, { css } from 'styled-components';

type Props = { children: ReactNode; viewBox?: string } & IconProps;

const SVGWrapper = styled.div<{ size: number }>`
  ${({ size }) =>
    size &&
    css`
      width: ${size}px;
      height: ${size}px;
    `};
`;

export const IconBase = forwardRef<SVGSVGElement, Props>(
  ({ size, color, viewBox, children, className }, ref) => {
    const iconSize = size && size > 24 ? size + 2 : 23;
    return (
      <SVGWrapper size={iconSize} className={className}>
        <svg
          ref={ref}
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
          width="100%"
          height="100%"
          color={color && `var(--amino-${color})`}
          viewBox={viewBox || `0 0 24 24`}
          preserveAspectRatio="XMaxYMax meet"
        >
          {children}
        </svg>
      </SVGWrapper>
    );
  }
);
