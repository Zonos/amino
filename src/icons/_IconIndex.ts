/** Common */
export { AIDuotoneIcon } from './AIDuotoneIcon';
export { AIIcon } from './AIIcon';
export { ArrowDownDuotoneIcon } from './ArrowDownDuotoneIcon';
export { ArrowDownIcon } from './ArrowDownIcon';
export { ArrowLeftDuotoneIcon } from './ArrowLeftDuotoneIcon';
export { ArrowLeftIcon } from './ArrowLeftIcon';
export { ArrowRightDuotoneIcon } from './ArrowRightDuotoneIcon';
export { ArrowRightIcon } from './ArrowRightIcon';
export { ArrowSwapDuotoneIcon } from './ArrowSwapDuotoneIcon';
export { ArrowSwapIcon } from './ArrowSwapIcon';
export { ArrowUpDuotoneIcon } from './ArrowUpDuotoneIcon';
export { ArrowUpIcon } from './ArrowUpIcon';
export { BagDuotoneIcon } from './BagDuotoneIcon';
export { BagIcon } from './BagIcon';
export { BankDuotoneIcon } from './BankDuotoneIcon';
export { BankIcon } from './BankIcon';
export { BellDuotoneIcon } from './BellDuotoneIcon';
export { BellIcon } from './BellIcon';
export { BellOffDuotoneIcon } from './BellOffDuotoneIcon';
export { BellOffIcon } from './BellOffIcon';
export { BookDuotoneIcon } from './BookDuotoneIcon';
export { BookIcon } from './BookIcon';
export { BookmarkAddDuotoneIcon } from './BookmarkAddDuotoneIcon';
export { BookmarkAddIcon } from './BookmarkAddIcon';
export { BookmarkCheckDuotoneIcon } from './BookmarkCheckDuotoneIcon';
export { BookmarkCheckIcon } from './BookmarkCheckIcon';
export { BookmarkDuotoneIcon } from './BookmarkDuotoneIcon';
export { BookmarkIcon } from './BookmarkIcon';
export { BookmarkOffDuotoneIcon } from './BookmarkOffDuotoneIcon';
export { BookmarkOffIcon } from './BookmarkOffIcon';
export { BookmarkRemoveDuotoneIcon } from './BookmarkRemoveDuotoneIcon';
export { BookmarkRemoveIcon } from './BookmarkRemoveIcon';
export { BookmarkSubtractDuotoneIcon } from './BookmarkSubtractDuotoneIcon';
export { BookmarkSubtractIcon } from './BookmarkSubtractIcon';
export { BoxesDuotoneIcon } from './BoxesDuotoneIcon';
export { BoxesIcon } from './BoxesIcon';
export { BufferDuotoneIcon } from './BufferDuotoneIcon';
export { BufferIcon } from './BufferIcon';
export { CalculatorDuotoneIcon } from './CalculatorDuotoneIcon';
export { CalculatorIcon } from './CalculatorIcon';
export { CalendarDuotoneIcon } from './CalendarDuotoneIcon';
export { CalendarIcon } from './CalendarIcon';
export { CardDuotoneIcon } from './CardDuotoneIcon';
export { CardIcon } from './CardIcon';
export { CaretDownDuotoneIcon } from './CaretDownDuotoneIcon';
export { CaretDownIcon } from './CaretDownIcon';
export { CaretLeftDuotoneIcon } from './CaretLeftDuotoneIcon';
export { CaretLeftIcon } from './CaretLeftIcon';
export { CaretRightDuotoneIcon } from './CaretRightDuotoneIcon';
export { CaretRightIcon } from './CaretRightIcon';
export { CaretUpDuotoneIcon } from './CaretUpDuotoneIcon';
export { CaretUpIcon } from './CaretUpIcon';
export { CartDuotoneIcon } from './CartDuotoneIcon';
export { CartIcon } from './CartIcon';
export { ChartDuotoneIcon } from './ChartDuotoneIcon';
export { ChartIcon } from './ChartIcon';
export { CheckCircleDuotoneIcon } from './CheckCircleDuotoneIcon';
export { CheckCircleIcon } from './CheckCircleIcon';
export { CheckmarkDuotoneIcon } from './CheckmarkDuotoneIcon';
export { CheckmarkIcon } from './CheckmarkIcon';
export { CheckoutDuotoneIcon } from './CheckoutDuotoneIcon';
export { CheckoutIcon } from './CheckoutIcon';
export { ChevronDownCircleDuotoneIcon } from './ChevronDownCircleDuotoneIcon';
export { ChevronDownCircleIcon } from './ChevronDownCircleIcon';
export { ChevronDownDuotoneIcon } from './ChevronDownDuotoneIcon';
export { ChevronDownIcon } from './ChevronDownIcon';
export { ChevronLeftCircleDuotoneIcon } from './ChevronLeftCircleDuotoneIcon';
export { ChevronLeftCircleIcon } from './ChevronLeftCircleIcon';
export { ChevronLeftDuotoneIcon } from './ChevronLeftDuotoneIcon';
export { ChevronLeftIcon } from './ChevronLeftIcon';
export { ChevronRightCircleDuotoneIcon } from './ChevronRightCircleDuotoneIcon';
export { ChevronRightCircleIcon } from './ChevronRightCircleIcon';
export { ChevronRightDuotoneIcon } from './ChevronRightDuotoneIcon';
export { ChevronRightIcon } from './ChevronRightIcon';
export { ChevronUpCircleDuotoneIcon } from './ChevronUpCircleDuotoneIcon';
export { ChevronUpCircleIcon } from './ChevronUpCircleIcon';
export { ChevronUpDuotoneIcon } from './ChevronUpDuotoneIcon';
export { ChevronUpIcon } from './ChevronUpIcon';
export { CircleDuotoneIcon } from './CircleDuotoneIcon';
export { CircleIcon } from './CircleIcon';
export { ClassifyDuotoneIcon } from './ClassifyDuotoneIcon';
export { ClassifyIcon } from './ClassifyIcon';
export { ClearDuotoneIcon } from './ClearDuotoneIcon';
export { ClearIcon } from './ClearIcon';
export { ClockDuotoneIcon } from './ClockDuotoneIcon';
export { ClockIcon } from './ClockIcon';
export { CodeCircleDuotoneIcon } from './CodeCircleDuotoneIcon';
export { CodeCircleIcon } from './CodeCircleIcon';
export { CodeDuotoneIcon } from './CodeDuotoneIcon';
export { CodeIcon } from './CodeIcon';
export { CoinsDuotoneIcon } from './CoinsDuotoneIcon';
export { CoinsIcon } from './CoinsIcon';
export { CollectDuotoneIcon } from './CollectDuotoneIcon';
export { CollectIcon } from './CollectIcon';
export { CommentDuotoneIcon } from './CommentDuotoneIcon';
export { CommentIcon } from './CommentIcon';
export { CopilotDuotoneIcon } from './CopilotDuotoneIcon';
export { CopilotIcon } from './CopilotIcon';
export { CopyDuotoneIcon } from './CopyDuotoneIcon';
export { CopyIcon } from './CopyIcon';
export { CubeDuotoneIcon } from './CubeDuotoneIcon';
export { CubeIcon } from './CubeIcon';
export { DashboardDuotoneIcon } from './DashboardDuotoneIcon';
export { DashboardIcon } from './DashboardIcon';
export { DiamondDuotoneIcon } from './DiamondDuotoneIcon';
export { DiamondIcon } from './DiamondIcon';
export { DislikeDuotoneIcon } from './DislikeDuotoneIcon';
export { DislikeIcon } from './DislikeIcon';
export { DocsDuotoneIcon } from './DocsDuotoneIcon';
export { DocsIcon } from './DocsIcon';
export { DollarDuotoneIcon } from './DollarDuotoneIcon';
export { DollarIcon } from './DollarIcon';
export { DoubleChevronDuotoneIcon } from './DoubleChevronDuotoneIcon';
export { DoubleChevronIcon } from './DoubleChevronIcon';
export { DragDuotoneIcon } from './DragDuotoneIcon';
export { DragIcon } from './DragIcon';
export { EditDuotoneIcon } from './EditDuotoneIcon';
export { EditIcon } from './EditIcon';
export { ExclamationMarkDuotoneIcon } from './ExclamationMarkDuotoneIcon';
export { ExclamationMarkIcon } from './ExclamationMarkIcon';
export { ExportDuotoneIcon } from './ExportDuotoneIcon';
export { ExportIcon } from './ExportIcon';
export { ExternalDuotoneIcon } from './ExternalDuotoneIcon';
export { ExternalIcon } from './ExternalIcon';
export { EyeDuotoneIcon } from './EyeDuotoneIcon';
export { EyeIcon } from './EyeIcon';
export { EyeOffDuotoneIcon } from './EyeOffDuotoneIcon';
export { EyeOffIcon } from './EyeOffIcon';
export { FileDuotoneIcon } from './FileDuotoneIcon';
export { FileIcon } from './FileIcon';
export { FileUploadDuotoneIcon } from './FileUploadDuotoneIcon';
export { FileUploadIcon } from './FileUploadIcon';
export { FilterDuotoneIcon } from './FilterDuotoneIcon';
export { FilterIcon } from './FilterIcon';
export { FolderListDuotoneIcon } from './FolderListDuotoneIcon';
export { FolderListIcon } from './FolderListIcon';
export { FolderUploadDuotoneIcon } from './FolderUploadDuotoneIcon';
export { FolderUploadIcon } from './FolderUploadIcon';
export { FusionDuotoneIcon } from './FusionDuotoneIcon';
export { FusionIcon } from './FusionIcon';
export { GlobeDuotoneIcon } from './GlobeDuotoneIcon';
export { GlobeIcon } from './GlobeIcon';
export { GraphQLDuotoneIcon } from './GraphQLDuotoneIcon';
export { GraphQLIcon } from './GraphQLIcon';
export { GridDuotoneIcon } from './GridDuotoneIcon';
export { GridIcon } from './GridIcon';
export { HelloDuotoneIcon } from './HelloDuotoneIcon';
export { HelloIcon } from './HelloIcon';
export { HelpDuotoneIcon } from './HelpDuotoneIcon';
export { HelpIcon } from './HelpIcon';
export { HexagonDuotoneIcon } from './HexagonDuotoneIcon';
export { HexagonIcon } from './HexagonIcon';
export { HomeDuotoneIcon } from './HomeDuotoneIcon';
export { HomeIcon } from './HomeIcon';
export { ImageDuotoneIcon } from './ImageDuotoneIcon';
export { ImageIcon } from './ImageIcon';
export { ImportDuotoneIcon } from './ImportDuotoneIcon';
export { ImportIcon } from './ImportIcon';
export { InfoDuotoneIcon } from './InfoDuotoneIcon';
export { InfoIcon } from './InfoIcon';
export { IntegrationDuotoneIcon } from './IntegrationDuotoneIcon';
export { IntegrationIcon } from './IntegrationIcon';
export { LandedCostDuotoneIcon } from './LandedCostDuotoneIcon';
export { LandedCostIcon } from './LandedCostIcon';
export { LaptopDuotoneIcon } from './LaptopDuotoneIcon';
export { LaptopIcon } from './LaptopIcon';
export { LeftPanelClosedDuotoneIcon } from './LeftPanelClosedDuotoneIcon';
export { LeftPanelClosedIcon } from './LeftPanelClosedIcon';
export { LeftPanelOpenDuotoneIcon } from './LeftPanelOpenDuotoneIcon';
export { LeftPanelOpenIcon } from './LeftPanelOpenIcon';
export { LightDuotoneIcon } from './LightDuotoneIcon';
export { LightIcon } from './LightIcon';
export { LikeDuotoneIcon } from './LikeDuotoneIcon';
export { LikeIcon } from './LikeIcon';
export { LineDownDuotoneIcon } from './LineDownDuotoneIcon';
export { LineDownIcon } from './LineDownIcon';
export { LineUpDuotoneIcon } from './LineUpDuotoneIcon';
export { LineUpIcon } from './LineUpIcon';
export { LinkDuotoneIcon } from './LinkDuotoneIcon';
export { LinkIcon } from './LinkIcon';
export { LocationDuotoneIcon } from './LocationDuotoneIcon';
export { LocationIcon } from './LocationIcon';
export { LockDuotoneIcon } from './LockDuotoneIcon';
export { LockIcon } from './LockIcon';
export { LoginDuotoneIcon } from './LoginDuotoneIcon';
export { LoginIcon } from './LoginIcon';
export { LogoutDuotoneIcon } from './LogoutDuotoneIcon';
export { LogoutIcon } from './LogoutIcon';
export { MailDuotoneIcon } from './MailDuotoneIcon';
export { MailIcon } from './MailIcon';
export { MaximizeDuotoneIcon } from './MaximizeDuotoneIcon';
export { MaximizeIcon } from './MaximizeIcon';
export { MenuDuotoneIcon } from './MenuDuotoneIcon';
export { MenuIcon } from './MenuIcon';
export { MinimizeDuotoneIcon } from './MinimizeDuotoneIcon';
export { MinimizeIcon } from './MinimizeIcon';
export { MinusCircleDuotoneIcon } from './MinusCircleDuotoneIcon';
export { MinusCircleIcon } from './MinusCircleIcon';
export { MinusDuotoneIcon } from './MinusDuotoneIcon';
export { MinusIcon } from './MinusIcon';
export { MobileDuotoneIcon } from './MobileDuotoneIcon';
export { MobileIcon } from './MobileIcon';
export { MoneyDuotoneIcon } from './MoneyDuotoneIcon';
export { MoneyIcon } from './MoneyIcon';
export { MonitorDuotoneIcon } from './MonitorDuotoneIcon';
export { MonitorIcon } from './MonitorIcon';
export { NightDuotoneIcon } from './NightDuotoneIcon';
export { NightIcon } from './NightIcon';
export { PaletteDuotoneIcon } from './PaletteDuotoneIcon';
export { PaletteIcon } from './PaletteIcon';
export { PauseCircleDuotoneIcon } from './PauseCircleDuotoneIcon';
export { PauseCircleIcon } from './PauseCircleIcon';
export { PenDuotoneIcon } from './PenDuotoneIcon';
export { PenIcon } from './PenIcon';
export { PercentBadgeDuotoneIcon } from './PercentBadgeDuotoneIcon';
export { PercentBadgeIcon } from './PercentBadgeIcon';
export { PercentDuotoneIcon } from './PercentDuotoneIcon';
export { PercentIcon } from './PercentIcon';
export { PlaneDuotoneIcon } from './PlaneDuotoneIcon';
export { PlaneIcon } from './PlaneIcon';
export { PlayCircleDuotoneIcon } from './PlayCircleDuotoneIcon';
export { PlayCircleIcon } from './PlayCircleIcon';
export { PlayDuotoneIcon } from './PlayDuotoneIcon';
export { PlayIcon } from './PlayIcon';
export { PlusCircleDuotoneIcon } from './PlusCircleDuotoneIcon';
export { PlusCircleIcon } from './PlusCircleIcon';
export { PlusDuotoneIcon } from './PlusDuotoneIcon';
export { PlusIcon } from './PlusIcon';
export { PrepayDuotoneIcon } from './PrepayDuotoneIcon';
export { PrepayIcon } from './PrepayIcon';
export { PrinterDuotoneIcon } from './PrinterDuotoneIcon';
export { PrinterIcon } from './PrinterIcon';
export { RateDuotoneIcon } from './RateDuotoneIcon';
export { RateIcon } from './RateIcon';
export { ReceiptDuotoneIcon } from './ReceiptDuotoneIcon';
export { ReceiptIcon } from './ReceiptIcon';
export { RefreshDuotoneIcon } from './RefreshDuotoneIcon';
export { RefreshIcon } from './RefreshIcon';
export { RemoveCircleDuotoneIcon } from './RemoveCircleDuotoneIcon';
export { RemoveCircleIcon } from './RemoveCircleIcon';
export { RemoveDuotoneIcon } from './RemoveDuotoneIcon';
export { RemoveIcon } from './RemoveIcon';
export { RestrictDuotoneIcon } from './RestrictDuotoneIcon';
export { RestrictIcon } from './RestrictIcon';
export { RightPanelClosedDuotoneIcon } from './RightPanelClosedDuotoneIcon';
export { RightPanelClosedIcon } from './RightPanelClosedIcon';
export { RightPanelOpenDuotoneIcon } from './RightPanelOpenDuotoneIcon';
export { RightPanelOpenIcon } from './RightPanelOpenIcon';
export { RocketDuotoneIcon } from './RocketDuotoneIcon';
export { RocketIcon } from './RocketIcon';
export { ScanDuotoneIcon } from './ScanDuotoneIcon';
export { ScanIcon } from './ScanIcon';
export { ScreenDuotoneIcon } from './ScreenDuotoneIcon';
export { ScreenIcon } from './ScreenIcon';
export { SearchDuotoneIcon } from './SearchDuotoneIcon';
export { SearchIcon } from './SearchIcon';
export { SendDuotoneIcon } from './SendDuotoneIcon';
export { SendIcon } from './SendIcon';
export { SettingsDuotoneIcon } from './SettingsDuotoneIcon';
export { SettingsIcon } from './SettingsIcon';
export { ShoppingListDuotoneIcon } from './ShoppingListDuotoneIcon';
export { ShoppingListIcon } from './ShoppingListIcon';
export { ShoppingTagDuotoneIcon } from './ShoppingTagDuotoneIcon';
export { ShoppingTagIcon } from './ShoppingTagIcon';
export { SquareDuotoneIcon } from './SquareDuotoneIcon';
export { SquareIcon } from './SquareIcon';
export { StarDuotoneIcon } from './StarDuotoneIcon';
export { StarIcon } from './StarIcon';
export { StarOffDuotoneIcon } from './StarOffDuotoneIcon';
export { StarOffIcon } from './StarOffIcon';
export { StarsDuotoneIcon } from './StarsDuotoneIcon';
export { StarsIcon } from './StarsIcon';
export { SunnyDuotoneIcon } from './SunnyDuotoneIcon';
export { SunnyIcon } from './SunnyIcon';
export { TagDuotoneIcon } from './TagDuotoneIcon';
export { TagIcon } from './TagIcon';
export { TaxDuotoneIcon } from './TaxDuotoneIcon';
export { TaxIcon } from './TaxIcon';
export { ThreeDotDuotoneIcon } from './ThreeDotDuotoneIcon';
export { ThreeDotIcon } from './ThreeDotIcon';
export { TotalLandedCostDuotoneIcon } from './TotalLandedCostDuotoneIcon';
export { TotalLandedCostIcon } from './TotalLandedCostIcon';
export { TrashCanDuotoneIcon } from './TrashCanDuotoneIcon';
export { TrashCanIcon } from './TrashCanIcon';
export { TriangleDuotoneIcon } from './TriangleDuotoneIcon';
export { TriangleIcon } from './TriangleIcon';
export { TruckDuotoneIcon } from './TruckDuotoneIcon';
export { TruckIcon } from './TruckIcon';
export { UFODuotoneIcon } from './UFODuotoneIcon';
export { UFOIcon } from './UFOIcon';
export { UserDuotoneIcon } from './UserDuotoneIcon';
export { UserIcon } from './UserIcon';
export { UtahDuotoneIcon } from './UtahDuotoneIcon';
export { UtahIcon } from './UtahIcon';
export { WarningDuotoneIcon } from './WarningDuotoneIcon';
export { WarningIcon } from './WarningIcon';
/** Custom */
export { AvatarIcon } from './custom/AvatarIcon';
export { ZonosIcon } from './custom/ZonosIcon';
export { ZonosLogoIcon } from './custom/ZonosLogoIcon';
/** Product Icons */
export { CheckoutColorIcon } from './custom/products/CheckoutColorIcon';
export { ClassifyColorIcon } from './custom/products/ClassifyColorIcon';
export { ClearColorIcon } from './custom/products/ClearColorIcon';
export { CollectColorIcon } from './custom/products/CollectColorIcon';
export { CounterCompanionColorIcon } from './custom/products/CounterCompanionColorIcon';
export { DashboardColorIcon } from './custom/products/DashboardColorIcon';
export { HelloColorIcon } from './custom/products/HelloColorIcon';
export { InclusivePricingColorIcon } from './custom/products/InclusivePricingColorIcon';
export { LandedCostColorIcon } from './custom/products/LandedCostColorIcon';
export { PrepayColorIcon } from './custom/products/PrepayColorIcon';
export { RateColorIcon } from './custom/products/RateColorIcon';
export { RestrictColorIcon } from './custom/products/RestrictColorIcon';
export { ScreenColorIcon } from './custom/products/ScreenColorIcon';
export { TaxColorIcon } from './custom/products/TaxColorIcon';
