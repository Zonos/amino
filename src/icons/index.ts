export { XIcon } from './XIcon';
export { DropdownIcon } from './DropdownIcon';
export { ChevronLeftIcon } from './ChevronLeftIcon';
export { ChevronRightIcon } from './ChevronRightIcon';
export { ChevronDownIcon } from './ChevronDownIcon';
export { ChevronUpIcon } from './ChevronUpIcon';
export { PlusIcon } from './PlusIcon';
export { SearchIcon } from './SearchIcon';
export { CheckIcon } from './CheckIcon';
export { HomeIcon } from './HomeIcon';
export { CogIcon } from './CogIcon';
export { AdminIcon } from './AdminIcon';
