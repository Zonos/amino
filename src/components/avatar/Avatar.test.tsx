import React from 'react';
import * as renderer from 'react-test-renderer';

import { ArrowDownIcon } from 'src/icons/ArrowDownIcon';

import { DynamicIconAvatar } from './DynamicIconAvatar';
import { IconAvatar } from './IconAvatar';
import { ImageAvatar } from './ImageAvatar';
import { UserAvatar } from './UserAvatar';

describe('Avatar DOM snapshot tests', () => {
  it('UserAvatar matches snapshot', () => {
    const tree = renderer
      .create(<UserAvatar size="lg" shape="round" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('IconAvatar matches snapshot', () => {
    const tree = renderer
      .create(<IconAvatar icon={<ArrowDownIcon />} size="lg" shape="round" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('ImageAvatar matches snapshot', () => {
    const tree = renderer
      .create(
        <ImageAvatar
          imageUrl="https://upload.wikimedia.org/wikipedia/commons/2/23/Mountain_Goat%2C_Enchantments_Basin.jpg"
          size="lg"
          shape="round"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('DynamicIconAvatar matches snapshot', () => {
    const tree = renderer
      .create(
        <DynamicIconAvatar icon="ArrowDownIcon" size="lg" shape="round" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
