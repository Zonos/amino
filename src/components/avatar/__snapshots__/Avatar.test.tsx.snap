// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar DOM snapshot tests DynamicIconAvatar matches snapshot 1`] = `
<div
  className="sc-bczRLJ cesvjd"
  shape="round"
  size="lg"
>
  <svg
    fill="none"
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.707 13.879a1 1 0 0 1 1.415 1.414l-4.415 4.414a1 1 0 0 1-1.414 0l-4.414-4.414a1 1 0 1 1 1.414-1.414L11 16.586V5a1 1 0 1 1 2 0v11.586l2.707-2.707Z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Avatar DOM snapshot tests IconAvatar matches snapshot 1`] = `
<div
  className="sc-bczRLJ cesvjd"
  shape="round"
  size="lg"
>
  <svg
    fill="none"
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.707 13.879a1 1 0 0 1 1.415 1.414l-4.415 4.414a1 1 0 0 1-1.414 0l-4.414-4.414a1 1 0 1 1 1.414-1.414L11 16.586V5a1 1 0 1 1 2 0v11.586l2.707-2.707Z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Avatar DOM snapshot tests ImageAvatar matches snapshot 1`] = `
<div
  className="sc-bczRLJ ivqPzI"
  shape="round"
  size="lg"
/>
`;

exports[`Avatar DOM snapshot tests UserAvatar matches snapshot 1`] = `
<div
  className="sc-bczRLJ kdoqip"
  shape="round"
  size="lg"
/>
`;
